packages/core/src/components/document-component-demo/document-component.demo.tsx

```diff
@@ -217,70 +258,285 @@ export class DocumentComponentDemo implements ComponentInterface {
     );
   }
 
+  private intersectionObserver = new IntersectionObserver(
+    (entries) => {
+      for (const entry of entries) {
+        if (!(entry.target instanceof HTMLElement)) {
+          throw new Error("element not instanceof HTMLElement");
+        }
+
+        const observing = this.getObservingByElement(entry.target);
+        if (!observing) {
+          throw new Error("observing not found");
+        }
+
+        // Volgens mij zit in de viewer ObserveVisibilityDirective een bug: Zowel setVisible() als setInvisible() worden bij entry.isIntersecting === true aangeroepen.
+        if (entry.isIntersecting) {
+          this.setVisible(observing.documentComponent);
+        } else {
+          this.setInvisible(observing.documentComponent);
+        }
+      }
+    },
+    /*
+      - Een positieve margin werkt buiten de viewport.
+      - "100%" verlegt over de Y-as de intersection de volledige viewport hoogte naar onder en naar boven, en:
+      - verlegt over de X-as de intersection de volledige viewport breedte naar links en naar rechts.
+
+      Ofwel, de "renderport" is 3x de viewport en valt daarmee ruim buiten de viewport.
+    */
+    { rootMargin: "100%" }
+  );
+
+  // Array van gerenderde DocumentComponents. @State decorator zodat er bij wijzigingen een nieuwe render gaat.
+  @State()
+  private visible: DocumentComponent[] = [];
+
+  private isVisible(documentComponent: DocumentComponent) {
+    return this.visible.some((d) => d === documentComponent);
+  }
+
+  private setVisible(documentComponent: DocumentComponent) {
+    this.visible = [...this.visible, documentComponent];
+  }
+
+  private setInvisible(documentComponent: DocumentComponent) {
+    this.visible = this.visible.filter((d) => d !== documentComponent);
+  }
+
+  // Bewust één @State() decorator: Deze array wordt ongecontroleerd bijgewerkt in componentDidRender(), met @State() zou dit infinite render loops veroorzaken
+  private heightsCache: Array<{ documentComponent: DocumentComponent; height: number }> = [];
+
+  private setHeight(documentComponent: DocumentComponent, height: number) {
+    const cache = this.heightsCache.find(({ documentComponent: d }) => d === documentComponent);
+
+    if (!cache) {
+      this.heightsCache.push({ documentComponent, height });
+    } else if (cache.height < height) {
+      // Note: Dit is enigszins dubieus, want een Document Component kan ook in ingeklapte vorm weggepoetst worden. Dan wil je de actuele (kleinere) height.
```

Comment:
+ De hoogte wisselt in het Viewer Grid bij het aanpassen van de breedte van het paneel. In een smal paneel zal de hoogte van een document component hoger zijn dan op een breed paneel.